rotkehlchen/chain/evm/decoding/aave/v3/decoder.py

```diff
@@ -300,13 +338,51 @@ def _decode_interest(
         )
         return decoded_events
 
+    def _collateral_swap(self, context: 'DecoderContext') -> DecodingOutput:
+        """Decode a collateral swap event from aave.
+
+        This swapped event logs the underlying token swapped. At this point we have decoded
+        only the spend event and not the receive events. Since what the user receives is
+        the wrapped aave token we can't use the action item because we don't know which
+        aave token we will receive. The receive event is decoded in _decode_interest that happens
+        in the post decoding.
+        """
+        if context.tx_log.topics[0] != SWAPPED:
+            return DEFAULT_DECODING_OUTPUT
+
+        swapped_addr = bytes_to_address(context.tx_log.topics[1])
+        for event in context.decoded_events:
+            if (
+                event.asset.is_evm_token() and
+                event.amount == asset_normalized_value(
+                    amount=int.from_bytes(context.tx_log.data[0:32]),  # swapped amount
+                    asset=(resolved_token := event.asset.resolve_to_evm_token()),
+                ) and
+                any(token.address == swapped_addr for token in resolved_token.underlying_tokens or [])  # noqa: E501
+            ):
+                event.event_type = HistoryEventType.TRADE
+                event.event_subtype = HistoryEventSubType.SPEND
+                event.counterparty = CPT_AAVE_V3
+                event.notes = f'Swap {event.amount} {resolved_token.symbol} AAVE v3 collateral'
+                event.extra_data = {'received_addr': bytes_to_address(context.tx_log.topics[2])}  # used to decode the other leg of the trade. Is removed before saving the event in the db.  # noqa: E501
```

Comment:
The removal only happens if we can find it though right? It can still be that somehow the event is broken and we end up saving this alone in the DB. Which is not super terrible, or is it?